api/services/BoxOrdering.js

```diff
@@ -5,12 +5,8 @@ module.exports = {
     const pokemonMapById = keyBy(box.contents, 'id');
     const orderedList = [];
     _.forEach(box._orderedIds, id => {
-      if (_.isString(id) && id in pokemonMapById) {
-        if (!pokemonMapById[id]._markedForDeletion) {
-          orderedList.push(pokemonMapById[id]);
-        }
-      } else {
-        sails.log.warn(`Pokemon ${id} is in box ${box.id}, but is not on its ordered ID list.`);
+      if (_.isString(id) && _.has(pokemonMapById, id) && !pokemonMapById[id]._markedForDeletion) {
```

Comment:
This is basically the same thing as before, except that it combines the two `if` statements into one and removes the warning message. The warning message had the potential to give false-alarms if `box.contents`  had been filtered to omit deleted elements previously. This was a bug that I hadn't noticed until now, probably because the tests now involve uploading a bunch of things to the same box, etc.